astropy/modeling/fitting_parallel.py

```diff
@@ -0,0 +1,615 @@
+import os
+import traceback
+import warnings
+from copy import deepcopy
+from math import ceil, log10, prod
+
+import numpy as np
+
+import astropy.units as u
+from astropy.modeling.utils import _combine_equivalency_dict
+from astropy.wcs.wcsapi import BaseHighLevelWCS, BaseLowLevelWCS
+from astropy.wcs.wcsapi.wrappers import SlicedLowLevelWCS
+
+__all__ = ["parallel_fit_dask"]
+
+
+def _pixel_to_world_values_block(*pixel, wcs=None):
+    """
+    Convert a block of pixel values to world values using a WCS. This is for
+    use in map_blocks.
+    """
+    world = wcs.pixel_to_world_values(*pixel[::-1])[::-1]
+    world = np.array(world)
+    return world
+
+
+def _wcs_to_world_dask(wcs, shape, chunks=None):
+    """
+    Given a WCS and a data shape, return an iterable of dask arrays
+    representing the world coordinates of the array.
+    """
+    import dask.array as da
+
+    pixel = tuple([np.arange(size) for size in shape])
+    pixel_nd = da.meshgrid(*pixel, indexing="ij")
+    world = da.map_blocks(
+        _pixel_to_world_values_block,
+        *pixel_nd,
+        wcs=deepcopy(wcs),
+        new_axis=0,
+        chunks=(3,) + chunks,
```

Comment:
This was a mistake and would not prevent N-dimensional WCS from being passed, but would just make it not be optimally efficient. I have fixed this to use the actual number of dimensions instead of 3.